### ISSUE TITLE
6d transpose not supported by TFLite GPU delegate

### ISSUE THREAD

**Issue by vbogach** [tensorflow/tensorflow#56537]
**System information**
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Ubuntu 22.04 LTS
- TensorFlow installed from (source or binary): binary (pip)
- TensorFlow version (or github SHA if from source): 2.9.1

**Provide the text output from tflite_convert**

```
Some ops are not supported by the native TFLite runtime, you can enable TF kernels fallback using TF Select. See instructions: https://www.tensorflow.org/lite/guide/ops_select 
TF Select ops: Transpose
Details:
        tf.Transpose(tensor<1x2x2x7x7x384xf32>, tensor<6xi32>) -> (tensor<1x2x7x2x7x384xf32>) : {device = ""}

```

if SELECT_TF_OPS is enabled:

```
Subgraph#0 main(T#0) -> [T#2]
  Op#0 FlexTranspose(T#0, T#1) -> [T#2]
GPU COMPATIBILITY WARNING: Not supported custom op FlexTranspose

GPU COMPATIBILITY WARNING: Subgraph#0 has GPU delegate compatibility issues at nodes 0 with TFLite runtime version 2.9.1
```

**Standalone code to reproduce the issue** 
```python3
import tensorflow as tf
from tensorflow import keras

input = keras.Input(shape=(2,2,7,7,384), batch_size=1)
output = tf.transpose(a=input, perm=(0, 1, 3, 2, 4, 5))
model = keras.Model(inputs=input, outputs=output, name="bug_model")
model.summary()

converter = tf.lite.TFLiteConverter.from_keras_model(model)

converter.target_spec.supported_ops = [
    tf.lite.OpsSet.TFLITE_BUILTINS, # enable TensorFlow Lite ops.
]
tflite_model = converter.convert()
tf.lite.experimental.Analyzer.analyze(model_content=tflite_model, gpu_compatibility=True)

open("bug.tflite", "wb").write(tflite_model)
```

**Any other info / logs**
6D transform is a part of an image transformer model.

**Comment by mohantym** [comment]
Hi @sachinprasadhs ! Could you please look at this issue. Attached [2.8](https://colab.sandbox.google.com/gist/mohantym/3dd3a134b5599f89162c0403d468760a/git_56537.ipynb#scrollTo=buk_wTuA7Tya), [2.9](https://colab.sandbox.google.com/gist/mohantym/3b8dc959517833398ccbd3b3815e9aed/git_56537.ipynb#scrollTo=buk_wTuA7Tya) and [nightly ](https://colab.sandbox.google.com/gist/mohantym/24cd766907f2b108516d0924d8f6d530/git_56537.ipynb#scrollTo=KTgwCO8v6-gD)for reference.

**Comment by impjdi** [comment]
As mentioned in the tflite_convert's error & warning messages, 6D tensors are not really supported in TFLite; only through Flex delegate which pulls in (non-mobile) TF code.

My team mates were able to get away with a hack folding / unfolding tensors into lower ranks to be able to use the GPU delegate.  See whether you can do that.  When you do it naively, you will indeed end up with 6D tensors which is not TFLite-friendly, especially not TFLite GPU.

**Comment by PINTO0309** [comment]
I have tried to implement a way to avoid `Transpose` over 6D being converted to `FlexTranspose`. I even generated a random tensor in 8 dimensions 1000 times to make sure the values matched Numpy's Transpose perfectly.

The implementation is quite verbose. I don't know if this is a meaningful implementation or not. However, I have successfully converted the Swin Transformer.
https://github.com/PINTO0309/onnx2tf/issues/93
